Backend/src/main/java/Commands/UserCommands/GetUserDataPaginated.java

```diff
@@ -26,23 +28,22 @@ protected HttpResponseTypes doCommand() {
         try {
             String query = "FOR u IN users LIMIT @offset,@count RETURN u";
             Map<String, Object> bindVars = new HashMap<String,Object> ();
-            bindVars.put("count",message.getLimit());
-            bindVars.put("offset", message.getPage()*message.getLimit());
+            bindVars.put("count",message.getParameter("limit"));
+            bindVars.put("offset", (int) message.getParameter("page") * (int) message.getParameter("limit"));
             ArangoCursor<BaseDocument> cursor = arangoDB.db(dbName).query(query, bindVars, null, BaseDocument.class);

```

Comment:
Will there be an abstraction for using the arango cursor, if yes make it general so that you can execute any AQL query with it

Reply:
Yes in the next pull request i will add a general `findAll` method.